File: frontend/src/types/ask.ts
export interface ChatMessage {
  role: 'user' | 'assistant';
  content: string;
  sources?: AskSource[];
}

export interface AskAIRequest {
  question: string;
}

export interface AskSource {
  marker: string;
  source_type: string;
  source_id: number;
  chunk_key: string;
  score: number;
}

export interface AskAIResponse {
  answer: string;
  sources: AskSource[];
}

export interface AgentAskRequest {
  question: string;
  conversation_id?: string;
  model_provider?: string;
}

export interface AgentAskResponse {
  answer: string;
  sources: AskSource[];
  conversation_id: string;
  used_tools: string[];
  is_general_knowledge: boolean;
}

File: backend/schemas/ask.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional

class AskRequest(BaseModel):
    question: str = Field(..., min_length=3, max_length=1000, description="The natural language question to ask.")
    model_provider: str = Field("gemini", description="The LLM provider to use (gemini, groq, ollama)")

class AskSource(BaseModel):
    marker: str
    source_type: str
    source_id: str
    chunk_key: str
    score: float

class AskResponse(BaseModel):
    answer: str
    sources: List[AskSource]
